src/host/output.cpp

```diff
@@ -102,12 +102,16 @@ static void _CopyRectangle(SCREEN_INFORMATION& screenInfo,
         auto sourcePos = source.GetWalkOrigin(walkDirection);
         auto targetPos = target.GetWalkOrigin(walkDirection);
 
+        // Note that we read two cells from the source before we start writing
+        // to the target, so a two-cell DBCS character can't accidentally delete
+        // itself when moving one cell horizontally.
+        auto next = OutputCell(*screenInfo.GetCellDataAt(sourcePos));
         do
         {
-            const auto data = OutputCell(*screenInfo.GetCellDataAt(sourcePos));
-            screenInfo.Write(OutputCellIterator({ &data, 1 }), targetPos);
-
+            const auto current = next;
             source.WalkInBounds(sourcePos, walkDirection);
+            next = OutputCell(*screenInfo.GetCellDataAt(sourcePos));

```

Comment:
Is it possible for this to result in `source` running out of space (somehow?) before target does? I guess we're doing it in the previous code as well, but maybe one fewer time.

Reply:
If you look at the `WalkInBounds` implementation, it just doesn't update the position once it has reached the end of the range, so all we'll be doing is reading the final position twice.

https://github.com/microsoft/terminal/blob/0eff8c06e32bff6c1e8d39e6505ef8a4c6d8ecbf/src/types/viewport.cpp#L405-L417